Additional star tree feedback. (#8650)
opensearch-project/documentation-website: _field-types/supported-field-types/star-tree.md

## Limitations

> The STIX feature has the following limitations:

Reviewer: We can either have all the limitations here or we can link up the limitations section in the other page.

Author: Let's move all the limitations here.

- A star-tree index should only be enabled on indexes whose data is not updated or deleted because standard updates and deletions are not accounted for in a star-tree index.
- Currently, only `one` star-tree index can be created per index. Support for multiple star-tree indexes will be added in a future version.

> 

Author: Works for me.

Define star-tree index mappings in the `composite` section in `mappings`. 

> The following example API request creates a corresponding star-tree index for `request_aggs`. To compute metric aggregations for `request_size` and `latency` fields with queries on `port` and `status` fields, configure the following mappings:

Reviewer: The following example API request creates a corresponding star-tree index named `request_aggs` for request aggregations

`request_aggs` is the star tree index name - maybe we can word it like above

}
```

## Star-tree index configuration options

You can customize your star-tree implementation using the following `config` options in the `mappings` section. These options cannot be modified without reindexing.

| Parameter  | Description   | 
| :--- | :--- |
| `ordered_dimensions` | A [list of fields](#ordered-dimensions) based on which metrics will be aggregated in a star-tree index. Required. | 
| `metrics` | A [list of metric](#metrics) fields required in order to perform aggregations. Required. |

> | `max_leaf_docs` | The maximum number of star-tree documents that a leaf node can point to. After the maximum number of documents is reached, child nodes will be created based on the unique values of the next field in the `ordered_dimension` (if any). Default is `10000`. A lower value will use more storage but result in faster query performance. Inversely, a higher value will use less storage but result in slower query performance. For more information, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure).  |

Reviewer 2: "unique values" => "unique value"?

> | `max_leaf_docs` | The maximum number of star-tree documents that a leaf node can point to. After the maximum number of documents is reached, child nodes will be created based on the unique values of the next field in the `ordered_dimension` (if any). Default is `10000`. A lower value will use more storage but result in faster query performance. Inversely, a higher value will use less storage but result in slower query performance. For more information, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure).  |

Author: ```suggestion
| `max_leaf_docs` | The maximum number of star-tree documents that a leaf node can point to. After the maximum number of documents is reached, child nodes will be created based on the unique value of the next field in the `ordered_dimension` (if any). Default is `10000`. A lower value will use more storage but result in faster query performance. Inversely, a higher value will use less storage but result in slower query performance. For more information, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure).  |
```

| `ordered_dimensions` | A [list of fields](#ordered-dimensions) based on which metrics will be aggregated in a star-tree index. Required. | 
| `metrics` | A [list of metrics](#metrics) fields required to perform aggregations. Required. |
| `max_leaf_docs` | The maximum number of STIX documents that a leaf node can point to. After the maximum number of documents is reached, children nodes will be created based on the unique vales of the next field in the `ordered_dimension` (if any). Default is `10000`. A lower value will use more storage but result in faster query performance. Inversely, a higher value will use less storage but result in slower query performance. For more information, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure).  |
| `skip_star_node_creation_for_dimensions`  | A list of dimensions for which STIX will skip star node creation. When `true`, this reduces storage size at the expense of query performance. Default is `false`. For more information about star nodes, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure). |

> 

Author: Yes. I'll do this once I've addressed your initial comments.

}
```

## STIX configuration options

> You can customize your STIX implementation using the following `config` options.

Author: ```suggestion
You can customize your STIX implementation using the following `config` options under the `mappings` section.
```

| Parameter  | Description   | 
| :--- | :--- |

> | `mappings` | A list of [mapping](#mapping-parameters) parameters. Required.

Reviewer: This is inaccurate, the config is part of mapping. And config contains 4 parameters , 2 simple ones max leaf docs and skip star node. And 2 complex ones, ordered dimensions and metrics.

If needed we can give a brief about ordered dimensions and metrics in the table here. And then we can continue with separate sections for ordered dimensions and metrics as already present

| `max_leaf_docs` | The maximum number of STIX documents that a leaf node can point to. After the maximum number of documents is reached, children nodes will be created based on the unique vales of the next field in the `ordered_dimension` (if any). Default is `10000`. A lower value will use more storage but result in faster query performance. Inversely, a higher value will use less storage but result in slower query performance. For more information, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure).  |
| `skip_star_node_creation_for_dimensions`  | A list of dimensions for which STIX will skip star node creation. When `true`, this reduces storage size at the expense of query performance. Default is `false`. For more information about star nodes, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure). |

> ### Mapping parameters

Reviewer: Same comment as above, these are all part of config. Lets not mention `mappings` . Please refer the associated json and we can word it on how they're structured

Author: Removed this section.

## Star-tree mapping parameters

> Specify any star-tree configuration mapping options in the `mappings` section. Parameters cannot be modified without reindexing documents.

Reviewer: Should we differentiate from mappings since its present a bit deeper in config section.

under corresponding STIX `config` section under `mappings`

 or something similar ?

Author: For sure, let me rework the order of this section.

### Metrics

> Configure any metric fields on which you need to perform aggregations. `Metrics` are required as part of a STIX configuration.

Author: ```suggestion
Configure any metric fields on which you need to perform aggregations. `Metrics` are required as part of a star-tree index configuration.
```

### Star-tree configuration parameters

The following `config` parameters are optional and cannot be modified following index creation.

| Parameter  | Description   | 
| :--- | :--- |

> | `max_leaf_docs` | The maximum number of STIX documents that a leaf node can point to. After the maximum number of documents is reached, children nodes of the leaf node will be created based on the unique `doc_value`. Default is `10000`. A lower value will use more storage but result in faster query performance. Inversely, a higher value will use less storage but result in slower query performance. For more information, see [Star-tree indexing structure]({{site.url}}{{site.baseurl}}/search-plugins/star-tree-index/#star-tree-index-structure).  |

Reviewer: ```The maximum number of STIX documents that a leaf node can point to. Once the node crosses this threshold,  children nodes will be created based on the unique values of the next field in `ordered_dimension` (if any).```
